File: apps/api/app/schemas/approvals.py
from __future__ import annotations

import uuid
from datetime import date, datetime

from pydantic import BaseModel


class ApprovalCenterSummary(BaseModel):
    pending: int = 0
    approved: int = 0
    rejected: int = 0
    needs_revision: int = 0


class ApprovalCenterItem(BaseModel):
    id: uuid.UUID
    type: str
    title: str
    user_name: str
    user_email: str
    department: str | None = None
    status: str
    submitted_at: datetime | None = None
    business_date: date | None = None
    description: str
    action_url: str
    available_actions: list[str] = []


class ApprovalCenterResponse(BaseModel):
    summary: ApprovalCenterSummary
    items: list[ApprovalCenterItem]

File: apps/web/src/lib/api/approvals.ts
import apiClient from './client';

export interface ApprovalCenterItem {
  id: string;
  type: string;
  title: string;
  user_name: string;
  user_email: string;
  department: string | null;
  status: string;
  submitted_at: string | null;
  business_date: string | null;
  description: string;
  action_url: string;
  available_actions?: string[];
}

export interface ApprovalCenterResponse {
  summary: Record<string, number>;
  items: ApprovalCenterItem[];
}

export const approvalsApi = {
  list: async (params?: Record<string, string>) => {
    const response = await apiClient.get<ApprovalCenterResponse>('/approvals', { params });
    return response.data;
  },
};
